repository-azure/build.gradle

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ext {
+    azureStorageBlobVerison = "12.8.0"
+    azureStorageBlobBatchVerison = "12.6.0"
+    azureStorageOkHttpVerison = "1.3.2"
+}
+
+dependencies {
+    compile project(':repository-commons')
+
+    implementation("com.azure:azure-core-http-okhttp:$azureStorageOkHttpVerison") {
+        exclude group: "com.fasterxml.jackson.core", module: "jackson-core"
+        exclude group: "com.fasterxml.jackson.datatype", module: "jackson-datatype-jsr310"
+        exclude group: "com.fasterxml.jackson.dataformat", module: "jackson-dataformat-xml"
+    }
+    implementation ("com.azure:azure-storage-blob:$azureStorageBlobVerison") {
+        exclude group: "com.fasterxml.jackson.core", module: "jackson-core"
+        exclude group: "com.azure", module: "azure-core-http-netty"
+        exclude group: "com.fasterxml.jackson.datatype", module: "jackson-datatype-jsr310"
+        exclude group: "com.fasterxml.jackson.dataformat", module: "jackson-dataformat-xml"
+        exclude group: "io.netty", module: "netty-tcnative-boringssl"
+    }
+    implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jacksonDatabind") {
+        exclude group: "com.fasterxml.jackson.core", module: "jackson-core"
+        exclude group: "com.fasterxml.jackson.datatype", module: "jackson-datatype-jsr310"
+        exclude group: "com.fasterxml.jackson.dataformat", module: "jackson-dataformat-xml"
+    }
+    implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-xml:$jacksonDatabind") {
+        exclude group: "com.fasterxml.jackson.core", module: "jackson-core"
+        exclude group: "com.fasterxml.jackson.datatype", module: "jackson-datatype-jsr310"
+        exclude group: "com.fasterxml.jackson.dataformat", module: "jackson-dataformat-xml"
+    }
+    implementation("com.fasterxml.jackson.core:jackson-databind:$jacksonDatabind") {
+        exclude group: "com.fasterxml.jackson.core", module: "jackson-core"
+        exclude group: "com.fasterxml.jackson.datatype", module: "jackson-datatype-jsr310"
+        exclude group: "com.fasterxml.jackson.dataformat", module: "jackson-dataformat-xml"
+    }
```

Comment:
Are these used at compile time? Should we mark them `runtimeOnly` instead? In this case all the tricky exclusions won't be needed, AFAIK.

Reply:
ok will try ... this is done explicitly due to the ES classloader